framework/src/main/java/org/checkerframework/framework/util/FlowExpressionParseUtil.java

```diff
@@ -181,12 +248,23 @@ private static boolean isMemberSelect(String s, FlowExpressionContext context) {
                     array.first.first + "[" + array.first.second + "]", array.second.substring(1));
         }
 
-        Pattern memberSelectOfStringPattern = anchored("(" + STRING_REGEX + ")" + "\\.(.*)");
-        Matcher m = memberSelectOfStringPattern.matcher(s);
-        if (m.matches()) {
-            return Pair.of(m.group(1), m.group(2));
+        // parses member selects which are strings
+        try {
+            Expression e = parseExpression(s);
+            if (e.getClass().equals(MethodCallExpr.class)) {
+                MethodCallExpr methodCallExpr = (MethodCallExpr) e;
+                String mceString = methodCallExpr.toString();
+                String receiver = methodCallExpr.getScope().get().toString();
+                String remaining = mceString.substring(receiver.length(), mceString.length());
+                if (receiver.charAt(0) == '\"') {
+                    return Pair.of(receiver, remaining.substring(1));
+                }
+            }
+        } catch (ParseProblemException e) {
+
```

Comment:
Never catch an exception without doing anything.  At the very least, add a comment saying why this is acceptable.
Keep the scope of the `try` statement as small as possible.
If the only statement that can throw `ParseProblemException` is `parseExpression`, then make the `try` body only that statement.